backend/test_observer/controllers/test_results/test_results.py

```diff
@@ -0,0 +1,210 @@
+        else:
+            filters.append(Environment.name.ilike(f"%{environments}%"))
+
+    if test_cases:
+        if "," in test_cases:
+            test_case_list = parse_csv_values(test_cases)
+            filters.append(TestCase.name.in_(test_case_list))
+        else:
+            filters.append(TestCase.name.ilike(f"%{test_cases}%"))
+
+    if template_ids:
+        template_id_list = parse_csv_values(template_ids)
+        filters.append(TestCase.template_id.in_(template_id_list))
+
+    if issues:
+        issue_ids = parse_csv_ids(issues)
+        filters.append(
+            TestResult.id.in_(
+                select(IssueTestResultAttachment.test_result_id)
+                .join(IssueTestResultAttachment.issue)
+                .where(Issue.id.in_(issue_ids))
+            )
+        )
+
+    if from_date:
+        filters.append(TestExecution.created_at >= from_date)  # type: ignore
+
+    if until_date:
+        filters.append(TestExecution.created_at <= until_date)  # type: ignore
+
+    # Build the main query with window function for total count and proper eager loading
+    query = (
+        select(TestResult, func.count().over().label("total_count"))
+        .join(TestResult.test_execution)
+        .join(TestExecution.environment)
+        .join(TestExecution.artefact_build)
+        .join(ArtefactBuild.artefact)
+        .join(TestResult.test_case)
+        .options(
+            # Load all the related data that the response model needs
+            selectinload(TestResult.test_case),
+            selectinload(TestResult.test_execution).selectinload(
+                TestExecution.environment
+            ),
+            selectinload(TestResult.test_execution)
+            .selectinload(TestExecution.artefact_build)
+            .selectinload(ArtefactBuild.artefact),
+            selectinload(TestResult.issue_attachments),
+        )
+        .order_by(desc(TestExecution.created_at), desc(TestResult.id))
+        .offset(offset)
+        .limit(limit)
+    )
+
+    # Apply all filters
+    if filters:
+        query = query.where(and_(*filters))
+
+    result = db.execute(query).all()
+
+    if result:
+        test_results = [row[0] for row in result]
+        total_count = result[0][1]
+
+        enhanced_results = []
+        for test_result in test_results:
+            test_result_response = TestResultResponse.model_validate(test_result)
+            test_execution_response = TestExecutionResponse.model_validate(
+                test_result.test_execution
+            )
+            artefact_response = ArtefactResponse.model_validate(
+                test_result.test_execution.artefact_build.artefact
+            )
+            artefact_build_response = ArtefactBuildMinimalResponse.model_validate(
+                test_result.test_execution.artefact_build
+            )
+
+            # Create the context object with all the nested data
+            enhanced_result = TestResultResponseWithContext(
```

Comment:
Nitpick but using validation alias with AliasPath you can specify where these fields should be found

```
    artefact_build: ArtefactBuildMinimalResponse = Field(
        validation_alias=AliasPath("test_result", "test_execution", "artefact_build")
    )
```
(like in the `controllers/issues/models.py`)